### DISCUSSION TITLE
Update to support RN >60.0?

### DISCUSSION THREAD

**Discussion by esegebart** [LunatiqueCoder/react-native-media-console#9] [Q&A] [answered]
I noticed you are only supporting RN < 45 right now. Will you be updating this to support newer versions? This library seems to be exactly what I need but I am on RN 68.2.

**Comment by LunatiqueCoder** [comment, marked as answer]
Hello @esegebart.

It should work with the latest versions, including yours. That part of the docs has not been updated. Please try it and let me know if you have any issues.

**Reply by esegebart** [reply]
AMAZING. Thank you so much. I am trying it right now and can't seem to get the Video to play so I will need to do some tinkering with it! This is like everything I have needed :)

**Reply by esegebart** [reply]
@criszz77 I was wondering do you have project where you have used react-native-media-console that I could look at? I am having a bit of trouble getting my video to play and wanted to see how you might have it set up to see if I am missing something. And how you have the onPlay set up to do something. :)

**Reply by LunatiqueCoder** [reply]
@esegebart With this snippet, the video should play automatically. If not, maybe you should double check if all [react-native-video installation steps](https://github.com/react-native-video/react-native-video/blob/master/API.md#installation) have been completed.

```javascript
import React, { useEffect, useState } from 'react';
import VideoPlayer  from 'react-native-media-console';
import { OnProgressData, OnSeekData } from 'react-native-video';

const App = () => {
  const [currentTime, setCurrentTime] = useState<number>();
  const [paused, setPaused] = useState<boolean>();
  const [isVideoReady, setIsVideoReady] = useState<boolean>();

  const onLoad = () => setIsVideoReady(true);
  const onProgress = (data: OnProgressData) => setCurrentTime(data.currentTime);
  const onSeek = (data: OnSeekData) => setCurrentTime(data.seekTime);
  const onEnd = () => setPaused(true);

  useEffect(() => {
    if (paused) {
      // onPause logic here
    } else {
      // onPlay logic here
    }
    // dependency array: each time any of these values changes,
    // the logic in the function provided to useEffect runs
    // again with the updated values
  }, [paused]);

  return (
    <VideoPlayer
      onLoad={onLoad}
      onProgress={onProgress}
      onSeek={onSeek}
      onEnd={onEnd}
      paused={paused}
      source={{ uri: 'https://cdn.videvo.net/videvo_files/video/premium/2021-04/large_watermarked/210329_09_Bali_4k_005_preview.mp4' }}
    />
  );
};

export default App;

```

**Reply by esegebart** [reply]
@criszz77 Ah, my video is playing because I can hear sound, I just can't see the video. So strange. I wonder what's causing that.

**Reply by LunatiqueCoder** [reply]
@esegebart You can try to post your example here, maybe I can help you.

**Reply by esegebart** [reply]
@criszz77  Okay here is my VideoPlayer component. Thank you so much for looking at it!

```
import React, { useRef, useEffect, useState } from 'react';
import { StyleSheet, View, Platform, TVEventHandler, useTVEventHandler, Image, Text, TouchableWithoutFeedback, TouchableOpacity } from 'react-native';
import Icon from "react-native-vector-icons/FontAwesome";
import VideoPlayer from 'react-native-media-console';
import { OnProgressData, OnSeekData } from 'react-native-video';
import moment from 'moment';

const Video = (props) => {

    const DURATION_MS = props.broadcastData.actual_duration * 1000;
    // The video we will play on the player.

    const video = props.source;
    const videoRef = useRef(null);
    const status = props.broadcastData;
    const [videoLoaded, setVideoLoaded] = useState(false);
    const [duration, setDuration] = useState(0.0);
    const [currentTime, setCurrentTime] = useState(0.0);
    const [paused, setPaused] = useState(false);
    const [isVideoReady, setIsVideoReady] = useState(false);

    const onLoad = () => setIsVideoReady(true);
    const onProgress = (data: OnProgressData) => setCurrentTime(data.currentTime);
    const onSeek = (data: OnSeekData) => setCurrentTime(data.seekTime);
    const onEnd = () => setPaused(true);

    // useTVEventHandler((evt) => {
    //     if (evt && evt.eventType === 'playPause') {
    //       setPaused(!paused);
    //     }
    // });
        
    return (
        <VideoPlayer
            videoRef={videoRef}
            onLoad={onLoad}
            onProgress={onProgress}
            onSeek={onSeek}
            onEnd={onEnd}
            paused={paused}
            source={{ uri: props.source }}
            style={{
                top: 0,
                left: 0,
                bottom: 0,
                right: 0,
                position: "absolute",
                height: "100%",
                width: "100%",
            }}
        />
    );
};

export default Video;
```

**Reply by LunatiqueCoder** [reply]
@esegebart Nothing should be wrong with your code, but I think the problem is where you  actually **use** the videoplayer. You might need to set dimensions on the parent container:

```javascript

  return (
    <View style={{ flex: 1 }}> // add this here, or height with width.
      <Video {...videoProps} />
    </View>
  );
```

Pasting the code where you use your **Video** component can also be helpful.

Good luck!

**Reply by esegebart** [reply]
@criszz77  OHHHH! That's right - prior to this I had a View around it just like that! That makes total sense. Okay I will do some tinkering and see if I can get it figured out :) I appreciate your help.

**Reply by esegebart** [reply]
@criszz77 IT'S WORKING!!!! Oh my the controls are so nice. YAY! Do you happen to know if this works on tvOS / Android TV also? I am going to try it and see if the controls are focusable. This would be SO good to have on a TV app because there is like.. nothing available for controls. :) Great job on this!!!!

**Reply by LunatiqueCoder** [reply]
@esegebart You might not need a parent/container `<View/>` at all, since the videoplayer has the `containerStyle` prop and it wraps the `<Video/>` component in a `<View/>` already: https://github.com/criszz77/react-native-media-console/blob/master/src/VideoPlayer.tsx#L379-L380 . Yes, I know, the one of the containers can be removed from the source code.

I'm very happy that it works. 🎉 

>Do you happen to know if this works on tvOS / Android TV also? I am going to try it and see if the controls are focusable. This would be SO good to have on a TV app because there is like..

Judging by the [react-native-video documentation](https://github.com/react-native-video/react-native-video/blob/master/API.md#tvos-installation), it should also work with tvOS. I personally haven't tried TV apps at all, what you're working on sounds at least very interesting, but the videoplayer controls should also work on tvOS, since they're all just plain React Native components. 🤔

Let me know how that goes for you, I'd be happy to help you with this! Knowing that this library also supports tvOS will be very satisfying.

And please star this repository if it has helped you 🙏

**Reply by esegebart** [reply]
I absolutely will let you know how it goes! I am using the react-native-tvOS fork since it changes the Pressables to be able to handle remote events. And will definitely star this repository :)

**Reply by esegebart** [reply]
@criszz77 Also, if there is anyway I can help contribute to this repository, please let me know! Whether it is just updating docs or small fixes, I am not sure I have enough experience to make large changes, but I would love to be able to give back somehow! I would like to learn more about contributing to open source.

**Reply by LunatiqueCoder** [reply]
@esegebart Help is always welcomed. Contributing to open source can also be very fun 🤩 

As an easy start, [line 41](https://github.com/criszz77/react-native-media-console/blob/master/README.md?plain=1#L41) from the README file which creates confusion can be removed. You might be able to do this directly from Github Web.

Somewhere in Octomber I'll probably create a roadmap with new features and fixes. Until then, feel free to try the library out. You can open an issue if you find any bugs or if you have any suggestions and I'll guide you to open a PR. Any kind of feedback is also very helpful.

Thank you!

**Reply by esegebart** [reply]
@criszz77 Awesome! :) I am excited about it! If this library can work with the react-native-tvos fork and get remote controls working with it - this would be the de facto media controls for all OTT (TV) apps because there aren't any other media controls packages that work with it, other than making a highly customizable component (which isn't very beginner friendly). And since yours is already super customizable - this would be an amazing direction for this repo to be able to support! I am currently building a Firestick app (android based) - and will be making tvOS and androidTV.
